src/SocketHttpServer.php

```diff
@@ -195,17 +215,7 @@ public function start(RequestHandler $requestHandler, ErrorHandler $errorHandler
             $this->logger->warning("The 'xdebug' extension is loaded, which has a major impact on performance.");
         }
 
-        if ($this->compressionMiddleware) {
-            if (!\extension_loaded('zlib')) {
-                $this->logger->warning(
-                    "The zlib extension is not loaded which prevents using compression. " .
-                    "Either activate the zlib extension or disable compression in the server's options."
-                );
```

Comment:
Ah, that should have been moved to the static constructors.

Reply:
I've swapped the $compressionMiddleware parameter for a boolean $enableCompression, which shows the warning now instead of throwing from the `CompressionMiddleware` constructor.